client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/RetryOnFailureOperation.java

```diff
@@ -107,7 +107,11 @@ protected boolean shouldRetry(int retryCount) {
    protected int logTransportErrorAndThrowExceptionIfNeeded(int i, HotRodClientException e) {
       String message = "Exception encountered. Retry %d out of %d";
       if (i >= transportFactory.getMaxRetries() || transportFactory.getMaxRetries() < 0) {
-         if (!triedCompleteRestart) {
+         if (transportFactory.trySwitchCluster(cacheName)){
+            log.debug("Switched to a different cluster");
```

Comment:
Since switching to a different cluster is quite a big thing, I suggest turning this into an info and promoting it to Log.java complete with the name of the cluster